core-plugins/src/main/java/co/cask/hydrator/plugin/batch/file/AbstractMetadataInputSplit.java

```
@@ -0,0 +1,162 @@
+
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Abstract class that implements information for InputSplit.
+ * Contains a list of fileMetadata that is assigned to the specific split.
+ */
+public abstract class AbstractMetadataInputSplit extends InputSplit implements Writable, Comparable {
+  protected List<AbstractFileMetadata> fileMetaDataList;
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractMetadataInputSplit.class);
+
+  public AbstractMetadataInputSplit(List<AbstractFileMetadata> fileMetaDataList) {
+    this.fileMetaDataList = fileMetaDataList;
+  }
+
+  public AbstractMetadataInputSplit() {
+    this.fileMetaDataList = new ArrayList<>();
+  }
+
+  public List<AbstractFileMetadata> getFileMetaDataList() {
+    return this.fileMetaDataList;
+  }
+
+  @Override
+  public void write(DataOutput dataOutput) throws IOException {
+    try {
+      // write obj summaries
+      dataOutput.writeLong(this.getLength());
+      for (int i = 0; i < this.getLength(); i++) {
+        // convert each filestatus (serializable) to byte array
+        AbstractFileMetadata fileMetaData = fileMetaDataList.get(i);
+
+        dataOutput.writeUTF(fileMetaData.getFileName());
+        dataOutput.writeUTF(fileMetaData.getFullPath());
+        dataOutput.writeLong(fileMetaData.getTimeStamp());
+        dataOutput.writeUTF(fileMetaData.getOwner());
+        dataOutput.writeLong(fileMetaData.getFileSize());
+        dataOutput.writeBoolean(fileMetaData.getIsFolder());
+        dataOutput.writeUTF(fileMetaData.getBasePath());
+        dataOutput.writeShort(fileMetaData.getPermission());
+
+        writeCredentials(dataOutput, fileMetaData.getCredentials());
+      }
+
+    } catch (Exception e) {
```

shouldn't log and skip. Just let the exception propagate, it should be handled by mapreduce.